include/api.php

```diff
@@ -1661,6 +1661,71 @@ function api_statuses_public_timeline($type)
 /// @TODO move to top of file or somewhere better
 api_register_func('api/statuses/public_timeline', 'api_statuses_public_timeline', true);
 
+/**
+ * @brief Returns the list of public federated posts this node knows about
+ *
+ * @param string $type Return format: json, xml, atom, rss
+ * @return array|string
+ * @throws ForbiddenException
+ */
+function api_statuses_networkpublic_timeline($type)
+{
+	$a = get_app();
+
+	if (api_user() === false) {
+		throw new ForbiddenException();
+	}
+
+	$user_info = api_get_user($a);
+
+	$since_id        = x($_REQUEST, 'since_id')        ? $_REQUEST['since_id']        : 0;
+	$max_id          = x($_REQUEST, 'max_id')          ? $_REQUEST['max_id']          : 0;
+
+	// pagination
+	$count = x($_REQUEST, 'count') ? $_REQUEST['count']   : 20;
+	$page  = x($_REQUEST, 'page')  ? $_REQUEST['page']    : 1;
+	if ($page < 1) {
+		$page = 1;
+	}
+	$start = ($page - 1) * $count;
+
+	$sql_extra = '';
+	if ($max_id > 0) {
+		$sql_extra = 'AND `item`.`id` <= ' . intval($max_id);
```

Comment:
You should always try to use conditions only on the first table in a "JOIN". Instead of `item`.`id` you could use `thread.`.`iid` - which holds the same data.